src/data/polynomial/degree/reverse_mul.lean

```diff
@@ -0,0 +1,605 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import tactic
+import data.polynomial.degree.basic
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+lemma nonempty_of_card_nonzero {α : Type*} {s : finset α} : s.card ≠ 0 → s.nonempty :=
+begin
+  contrapose!,
+  rw [nonempty_iff_ne_empty, not_not, card_eq_zero, imp_self],
+  trivial,
+end
+
+lemma nonempty_of_card_le_two {α : Type*} {s : finset α} (H : 2 ≤ s.card) : s.nonempty :=
+begin
+  apply nonempty_of_card_nonzero,
+  intro,
+  rw [a, nat.le_zero_iff] at H,
+  revert H,
+  exact two_ne_zero,
+end
+
+lemma card_two {sup : finset ℕ} (H : 2 ≤ sup.card) : finset.min' sup ≠ finset.max' sup :=
+mt (λ h, congr_fun h _) (ne_of_lt (finset.min'_lt_max'_of_card sup H))
+
+lemma card_two_d {sup : finset ℕ} (H : 2 ≤ sup.card) : finset.min' sup (nonempty_of_card_le_two H) ≠ finset.max' sup (nonempty_of_card_le_two H) :=
+begin
+  apply ne_of_lt,
+  apply finset.min'_lt_max'_of_card,
+  exact nat.succ_le_iff.mp H,
+end
+
+lemma coeff_eq_zero_of_not_mem_support {a : ℕ} : a ∉ f.support → f.coeff a = 0 :=
+begin
+  contrapose,
+  push_neg,
+  exact mem_support_iff_coeff_ne_zero.mpr,
+end
+
+lemma leading_coeff_nonzero_of_nonzero : f ≠ 0 ↔ leading_coeff f ≠ 0 :=
+begin
+  exact not_congr leading_coeff_eq_zero.symm,
+end
+
+lemma mem_support_term {n a : ℕ} {c : R} : a ∈ (C c * X ^ n).support → a = n :=
+begin
+  intro,
+  rw [mem_support_iff_coeff_ne_zero, coeff_C_mul_X c n a] at a_1,
+  finish,
+end
+
+lemma support_term (c : R) (n : ℕ) : (C c * X ^ n).support ⊆ singleton n :=
+begin
+  intro a,
+  rw mem_singleton,
+  exact mem_support_term,
+end
+
+lemma card_support_term_le_one {c : R} {n : ℕ} : (C c * X ^ n).support.card ≤ 1 :=
+begin
+  rw ← card_singleton n,
+  apply card_le_of_subset,
+  exact support_term c n,
+end
+
+lemma nat_degree_mem_support_of_nonzero : f ≠ 0 → f.nat_degree ∈ f.support :=
+begin
+  intro,
+  apply (f.3 f.nat_degree).mpr,
+  exact leading_coeff_nonzero_of_nonzero.mp a,
+end
+
+lemma le_nat_degree_of_mem_supp (a : ℕ) :
+  a ∈ f.support → a ≤ nat_degree f:=
+begin
+  rw mem_support_iff_coeff_ne_zero,
+  exact le_nat_degree_of_ne_zero,
+end
+
+lemma le_degree_of_mem_supp (a : ℕ) :
+  a ∈ f.support → a ≤ nat_degree f :=
+begin
+  rw mem_support_iff_coeff_ne_zero,
+  exact le_nat_degree_of_ne_zero,
+end
+
+lemma non_zero_of_nonempty_support : f.support.nonempty → f ≠ 0 :=
+begin
+  intro,
+  cases a with N Nhip,
+  rw mem_support_iff_coeff_ne_zero at Nhip,
+  finish,
+end
+
+lemma nonempty_support_of_non_zero : f ≠ 0 → f.support.nonempty :=
+begin
+  intro fne,
+  rw nonempty_iff_ne_empty,
+  apply ne_empty_of_mem,
+  rw mem_support_iff_coeff_ne_zero,
+  exact leading_coeff_nonzero_of_nonzero.mp fne,
+end
+
+lemma non_zero_iff : f.support.nonempty ↔ f ≠ 0 :=
```

Comment:
Call this `support_nonempty_iff` or `nonempty_support_iff` instead. It's not the only interesting thing about a polynomial being nonzero :)
Also, do you need the two separate directions of this lemma if you have the `iff`?